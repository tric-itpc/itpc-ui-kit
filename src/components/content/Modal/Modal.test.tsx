import React from "react"

import { fireEvent, render } from "@testing-library/react"

import { Modal, ModalContent, ModalFooter } from "./index"

describe("Компонент Modal", () => {
  test("Снимок Modal с заголовком и контентом", () => {
    const { asFragment } = render(
      <Modal onClose={() => {}} title="Modal Title" isOpen>
        <p>Modal Content</p>
      </Modal>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снимок Modal без контента", () => {
    const { asFragment } = render(
      <Modal onClose={() => {}} title="Modal Title" isOpen />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снимок Modal с иконкой закрытия", () => {
    const { asFragment } = render(
      <Modal
        iconClose={<button>Close</button>}
        onClose={() => {}}
        title="Modal Title"
        isOpen
      >
        <p>Modal Content</p>
      </Modal>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Рендер Modal с заголовком и контентом", () => {
    const { container, getByText } = render(
      <Modal title="Заголовок" isOpen>
        <p>Контент</p>
      </Modal>
    )

    const modalHeader = getByText("Заголовок")
    expect(modalHeader).toBeInTheDocument()

    const modalContent = container.querySelector(".itpc-modal")
    expect(modalContent).toBeInTheDocument()
  })

  test("Закрытие Modal при клике на оверлей, если isOverlayClickable=true", () => {
    const onCloseMock = jest.fn()
    const { container } = render(
      <Modal onClose={onCloseMock} title="Заголовок" isOpen isOverlayClickable>
        <p>Контент</p>
      </Modal>
    )

    const overlay = container.querySelector(".itpc-modal-overlay")
    fireEvent.click(overlay!)

    expect(onCloseMock).toHaveBeenCalledTimes(1)
  })

  test("Не закрывается Modal при клике на оверлей, если isOverlayClickable=false", () => {
    const onCloseMock = jest.fn()
    const { container } = render(
      <Modal
        isOverlayClickable={false}
        onClose={onCloseMock}
        title="Заголовок"
        isOpen
      >
        <p>Контент</p>
      </Modal>
    )

    const overlay = container.querySelector(".itpc-modal-overlay")
    fireEvent.click(overlay!)

    expect(onCloseMock).not.toHaveBeenCalled()
  })

  test("Рендер Modal с иконкой закрытия", () => {
    const { container } = render(
      <Modal iconClose={<button>Закрыть</button>} title="Заголовок" isOpen>
        <p>Контент</p>
      </Modal>
    )

    const closeButton = container.querySelector("button")
    expect(closeButton).toBeInTheDocument()
  })

  test("Modal закрывается при изменении isOpen", () => {
    const { container, rerender } = render(<Modal title="Заголовок" isOpen />)
    expect(container.querySelector(".itpc-modal_opened")).toBeInTheDocument()

    rerender(<Modal isOpen={false} title="Заголовок" />)
    expect(container.querySelector(".itpc-modal_closed")).toBeInTheDocument()
  })
})

describe("Компонент ModalContent", () => {
  test("Рендер ModalContent с содержимым", () => {
    const { getByText } = render(
      <ModalContent>
        <p>Контент</p>
      </ModalContent>
    )

    const content = getByText("Контент")
    expect(content).toBeInTheDocument()
  })

  test("Присваивание дополнительного класса для ModalContent", () => {
    const { container } = render(<ModalContent className="custom-class" />)
    const content = container.querySelector(".itpc-modal__content")
    expect(content).toHaveClass("custom-class")
  })
})

describe("Компонент ModalFooter", () => {
  test("Рендер ModalFooter с содержимым", () => {
    const { getByText } = render(
      <ModalFooter>
        <p>Контент футера</p>
      </ModalFooter>
    )

    const footerContent = getByText("Контент футера")
    expect(footerContent).toBeInTheDocument()
  })

  test("Присваивание дополнительного класса для ModalFooter", () => {
    const { container } = render(<ModalFooter className="custom-class" />)
    const footer = container.querySelector(".itpc-modal__footer")
    expect(footer).toHaveClass("custom-class")
  })
})
