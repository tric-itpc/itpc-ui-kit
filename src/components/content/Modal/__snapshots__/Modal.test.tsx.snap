// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Modal Снимок Modal без контента 1`] = `
<DocumentFragment>
  <div
    class="itpc-modal-overlay itpc-modal-overlay_opened"
  >
    <div
      class="itpc-modal itpc-modal_opened"
      id="itpc-modal"
    >
      <div
        class="itpc-modal__header"
      >
        Modal Title
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Modal Снимок Modal с заголовком и контентом 1`] = `
<DocumentFragment>
  <div
    class="itpc-modal-overlay itpc-modal-overlay_opened"
  >
    <div
      class="itpc-modal itpc-modal_opened"
      id="itpc-modal"
    >
      <div
        class="itpc-modal__header"
      >
        Modal Title
      </div>
      <p>
        Modal Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Modal Снимок Modal с иконкой закрытия 1`] = `
<DocumentFragment>
  <div
    class="itpc-modal-overlay itpc-modal-overlay_opened"
  >
    <div
      class="itpc-modal itpc-modal_opened"
      id="itpc-modal"
    >
      <div
        class="itpc-modal__header"
      >
        Modal Title
        <button>
          Close
        </button>
      </div>
      <p>
        Modal Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
